modules/bm_apps/examples/Makefile.am

```diff
@@ -1,4 +1,14 @@
 AM_CPPFLAGS = -I$(top_srcdir)/modules/bm_apps/include
+
+if P4THRIFT
+  AM_CPPFLAGS+= $(THRIFT_LIB)
```

Comment:
`CPPFLAGS` are the preprocessor flags...
You can add `$(THRIFT_LIB)` to `LDADD` below, and you don't need the `ifndef` block that follows